Add initial support for custom bar-id

mikkeloscar opened the pull request:
Sets a default bar-id on all bars. This can be changed with the `id` sub-command (once implemented).

SirCmpwn commented on sway/commands.c:
```diff
@@ -1124,9 +1124,26 @@ static struct cmd_results *cmd_bar(int argc, char **argv) {
 
+	// set bar id
+	int i, n;
+	int len = 5; // "bar-" + \0
```
Can you define this closer to where it's used, and add const?

mikkeloscar replied:
I have moved it down to the do-while loop, but I'm not sure what should be const?

SirCmpwn replied:
Nevermind, was me being stupid. I think there's a function in here somewhere that tells you how big a number is, we should reuse it.

SirCmpwn replied:
https://github.com/SirCmpwn/sway/blob/master/swaygrab/main.c#L16-L24

SirCmpwn replied:
Anyone want to take bets on when we'll get the first bug report for screens >1,000,000 pixels wide?

mikkeloscar replied:
`numlen` is defined in `swaygrab/main.c`, where should I put it for general use?

SirCmpwn replied:
`common/util.c` or something

mikkeloscar commented:
I have moved `numlen()` to `common/util.c` and merged it with `sway/util.c` which only had one function.

I have also updated the bar-id stuff to use `numlen`.

Hope it's ok to have all this in a single PR.

SirCmpwn commented:
It's fine for this to all be in one PR. Thanks for taking care of that.
